extensions-contrib/opentelemetry-emitter/src/main/java/org/apache/druid/emitter/opentelemetry/DruidContextTextMapGetter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.opentelemetry;
+
+import io.opentelemetry.context.propagation.TextMapGetter;
+
+import java.util.Map;
+
+public class DruidContextTextMapGetter implements TextMapGetter<Map<String, String>>
```

Comment:
Can we add a javadoc here? What makes this `TextMapGetter` specific to the `DruidContext` or is it a bit more general?

Reply:
Along these same lines, I'd recommend refactoring this to operate at a higher level of abstraction.  The true "carrier" for the text map is the Druid `ServiceMetricEvent`.  You're putting a lot of the logic of actually "getting the text map" in the `OpenTelemetryEmitter`.  

I'd move it here for better separation of concerns.

```java
public class DruidContextTextMapGetter implements TextMapGetter<ServiceMetricEvent>
{
  @Override
  public Iterable<String> keys(ServiceMetricEvent event)
  {
    return getContext(event).keySet();
  }

  @Nullable
  @Override
  public String get(@Nullable ServiceMetricEvent event, String key)
  {
    return getContext(event).get(key).toString();
  }

  @SuppressWarnings("unchecked")
  private Map<String, Object> getContext(ServiceMetricEvent event)
  {
    Object context = event.getUserDims().get("context");
    if (!(context instanceof Map)) {
      return Collections.emptyMap();
    }
    return (Map<String, Object>) context;
  }
}
```